db/records.py

```diff
@@ -72,6 +72,17 @@ def get_records(
                   field, in addition to an 'value' field if appropriate.
                   See: https://github.com/centerofci/sqlalchemy-filters#filters-format
     """
+    if not order_by:
+        # Set default ordering if none was requested
+        if len(table.primary_key.columns) > 0:
+            # If there are primary keys, order by all primary keys
+            order_by = [{'field': col, 'direction': 'asc'}
+                        for col in table.primary_key.columns]
```

Comment:
Under which conditions can `table.primary_key.columns` change order?  Is it only when the user themselves modifies the underlying primary key?  

I'm hoping that SQLAlchemy publishes the columns in the order they occur in the underlying index.  If so, that's great since we'll also get some performance advantages.  Otherwise, we should try to
- use the order of the columns in the tuples of the underlying index, or
- (if we can't find that order) use some canonical (say lexicographical) order for the columns when constructing the `order_by` list.

Reply:
I poked around the sqlalchemy source code a bit, and found [how they reflect the primary keys](https://github.com/sqlalchemy/sqlalchemy/blob/84be0b42e03b0aaf8000637c4b5e348a39333e2f/lib/sqlalchemy/dialects/postgresql/base.py#L3897). It seems to be they order the keys dependent on the order of the columns in the table. Assuming we don't plan on changing the order of columns, this seems like it should be alright? I have added a simple sort to the primary keys, which I'll revert if the default behavior seems reasonable.

Reply:
> It seems to be they order the keys dependent on the order of the columns in the table. 

This isn't quite correct. They're unnesting the actual index definition to grab the column `attnum`s, then using the `attnum` (the order in the table) to get the column names.  Note the line
```SQL
                ORDER BY k.ord
```
This is the part (along with the earlier call to `generate_subscripts`) that returns the columns in the order they're specified in the _index_, not in the table.

Since that's (IMO) the best order to use (it should be stable unless the table is completely reindexed, and using that order will be most efficient performance-wise), I think we should stick with the default order from SQLAlchemy rather than sorting it ourselves (so please revert that change).  Sorry for sending you on the wild goose-chase.  Thanks for finding the relevant snippet.